aiida_vasp/workchains/verify.py

```diff
@@ -91,7 +91,7 @@ def init_next_workchain(self):
         self.ctx.inputs.update(self.exposed_inputs(self._next_workchain))
 
         # Make sure we do not have any floating dict (convert to Dict)
-        self.ctx.inputs = prepare_process_inputs(self.ctx.inputs)
+        self.ctx.inputs = prepare_process_inputs(self.ctx.inputs, namespaces=['dynamics'])
```

Comment:
I think when we reach the verify workchain we should with the current code base have everything in `parameters`. Or?

Reply:
Okey, but we can just let it sit in parameters I think, right? Maybe it is better to keep the workchain clean of such inputs? If we ever make a dynamics workchain, or something related to molecular dynamics, things might look different. Do you agree?

Reply:
I just wonder how this would work, right now, what is being done is that the `prepare_process_inputs` sets the inputs from `dynamics` as part of the `parameters` that are then passed to the workchain during submission, this is then propagated to the `write_poscar` etc. If we remove the `namespaces`  call here, what will happen is that the `selective_dynamics` array will not be flattened and passed as a normal list to the parameters dictionary. This results in a serilization error, as it will try to store the `ArrayData` `dynamics`  inside a `Dict`.

I have no problem change it, just unsure on how to do it to avoid the serialization error. Do you have any suggestions?

Reply:
Aha, I did not think about this issue in fact. Good point. Can we use a list for the `selective_dynamics` flags from get go? I do not think this will amount to a very big array, which might be a reason to go with numpy, e.g. to save it in binary form? Or do we really need the numpy array? I do not see that as a requirement as we also do not do a lot of calculations with that array itself. It is mainly a container from the content to/from `POSCAR`. We might however do something more useful with it in the future, that I do not really know at this point. Thoughts?

Reply:
I think it is no problem in transforming it into a `List` I tested it and it works in the same way. The problem is that still I need to pass the `dynamics` namespace to the `prepare_process_inputs` as it will otherwise complain about `inputs` getting the wrong type of input
```
TypeError: port namespace `inputs.dynamics` received `<class 'aiida.orm.nodes.data.dict.Dict'>` instead of a dictionary
```

Reply:
Okey, thanks. I think this is due to the fact that it is not a regular dict, but a `Dict` type. I will try to run this branch and test a few things. Thanks a lot.

Reply:
Fantastic @espenfl . Let me know how it goes and if any more modifications are required please just let me know.

Reply:
@JPchico Yes, we need this absolutely. Consequence of allowing the dynamic input on the VASP workchain.